src/horreum/horreum_client.py

```diff
@@ -62,7 +62,12 @@ async def setup(self):
         """
 
         if self.__credentials:
-            if self.__credentials.username is not None:
+            if self.__client_config.auth_method == AuthMethod.API_KEY and self.__credentials.apikey is not None:
```

Comment:
I would refactor the `if` condition this way, mainly because you were relying on the fact that `self.__client_config` was not None but this is not true as it is optional
```suggestion
            if self.__credentials.apikey is not None and (self.__client_config is None or self.__client_config.auth_method == AuthMethod.API_KEY):
```